### PR TITLE
Patch constructor for async AzureAIClient

### PR DESCRIPTION
**Pull request by dargilco**
- Patch constructor for async AzureAIClient
- Add async operations for .inference and .endpoints
- Add missing close, __enter__ and __exit__ for the patch sync constructor
- Add samples

One of the async .inference samples still does not fully work, so I'm still investigating that, but I believe the new patched async construction for AzureAIClient should work, and should unblock async work on Agents and Evaluations.

**Note:** please use the `async with` pattern in your async sample code, per request from Johan. Otherwise Johan will not sign off on releasing them. See the examples I have in this PR.

### CODE REVIEW

**Review by howieleung — APPROVED** [review]

File: sdk/ai/azure-ai-client/azure/ai/client/aio/operations/_patch.py [hunk 1]
```diff
@@ -6,9 +6,216 @@
...
-__all__: List[str] = []  # Add all objects you want publicly available to users at this package level
+logger = logging.getLogger(__name__)
+
+
+class InferenceOperations:
+
+    def __init__(self, outer_instance):
+        self.outer_instance = outer_instance
+
+    async def get_chat_completions_client(self) -> "ChatCompletionsClient":
```
Comment (jhakulin) [inline sdk/ai/azure-ai-client/azure/ai/client/aio/operations/_patch.py:24]: get embedding client method is named as "get_async_embeddings_client". This one is "get_chat_completions_client", is there a reason being not consistent?

File: sdk/ai/azure-ai-client/samples/endpoints/async_samples/sample_endpoints_async.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        print(endpoint)
+
+        # Get an endpoint by its name:
+        endpoint = await ai_client.endpoints.get(
+            endpoint_name=os.environ["AI_CLIENT_CONNECTION_NAME"], populate_secrets=True  # Required.
+        )
+        print("====> Get endpoint by name:")
+        print(endpoint)
+
+        exit()
```
Comment (jhakulin) [inline sdk/ai/azure-ai-client/samples/endpoints/async_samples/sample_endpoints_async.py:74]: Why there is exit() here? The code below does not ever run if I am looking correctly?

File: sdk/ai/azure-ai-client/azure/ai/client/aio/operations/_patch.py (hunk 1 above)
Comment (dargilco) [reply sdk/ai/azure-ai-client/azure/ai/client/aio/operations/_patch.py:24]: Oh, forgot to change the name back to the original "get_embeddings_client". Thanks @jhakulin

File: sdk/ai/azure-ai-client/samples/endpoints/async_samples/sample_endpoints_async.py (hunk 2 above)
Comment (dargilco) [reply sdk/ai/azure-ai-client/samples/endpoints/async_samples/sample_endpoints_async.py:74]: Still need to polish the bottom half of this simple, it's work in progress (but should be functional). I'll take care of that in another PR.

File: sdk/ai/azure-ai-client/azure/ai/client/aio/operations/_patch.py (hunk 1 above)
Comment (jhakulin) [reply sdk/ai/azure-ai-client/azure/ai/client/aio/operations/_patch.py:24]: Should this "get_async_azure_openai_client" be also then without async?

**Review by jhakulin — APPROVED** [review]

File: sdk/ai/azure-ai-client/azure/ai/client/aio/operations/_patch.py [hunk 3]
```diff
@@ -6,9 +6,216 @@
...
+            logger.debug(
+                "[InferenceOperations.get_embeddings_client] Creating EmbeddingsClient using SAS authentication"
+            )
+            client = EmbeddingsClient(endpoint=endpoint.endpoint_url, credential=endpoint.token_credential)
+        else:
+            raise ValueError("Unknown authentication type")
+
+        return client
+
+    async def get_async_azure_openai_client(self) -> "AzureOpenAI":
```
[the remark this answers is not in the document]
Comment (dargilco) [reply sdk/ai/azure-ai-client/azure/ai/client/aio/operations/_patch.py:101]: yes, thanks! Fixed.

**Pull request merged**